### DISCUSSION TITLE
useProxy in react native?

### DISCUSSION THREAD

**Discussion by GollyJer** [pmndrs/valtio#311] [Q&A]
I have valtio working great with `useSnapshot`. But, I'd LOVE to be able to use `useProxy`.

I've followed all the instructions but when I try and use it I get an `unable to find "path"` error.
So I tried installing `path` from https://www.npmjs.com/package/path. And then got another error.

It seems maybe it's a React Native issue rather than a configuration error.

Is `useProxy` expected to work in React Native? Thanks.

**Comment by dai-shi** [comment]
Maybe not? `useProxy` uses babel-plugin-macros, and it only works where it's available.
Found this article: https://www.freecodecamp.org/news/using-babel-macros-with-react-native-8615aaf5b7df/
So, the config issue?

**Reply by GollyJer** [reply]
Thanks @dai-shi I had things set up the same way as that article.
For now I'll stick to useSnapshot and will update here when I get back it. For now I have other more important questions. 😀
